### ISSUE TITLE
Implement google CPC pipeline in v2

### ISSUE THREAD

**Issue by slifty** [prior-art-archive/overview#32]
This issue came out of conversation related to issue #4.

We don't have CPC code data in the dev database for v2.  The process is as follows:

1) Google accesses a site map on a regular basis and processes the site.
2) Google publishes a json file containing CPC codes related to each page.

In order to enable this pipeline we need to set up the URL route that google pings every night to ask for a site map.  The URL that google expects the sitemap to be posted to is predetermined: `/cpc/sitemap`

Some relevant files for this issue:

1. [Here's an example](https://s3.amazonaws.com/prior-art-archive-sftp/_priorArtArchive/sitemap.txt) of a sitemap google would parse.

2. [The code](https://github.com/prior-art-archive/api-v1/blob/master/scripts/buildSiteMap.js) that generates a sitemap doesn't exist in this version yet.

3. [The output from google](https://patentimages.storage.googleapis.com/cpc_feedback/cpc_results.txt) which will always live in the same URL and needs to be ingested on a regular basis.

The json includes our primary keys, so mapping Google's output to our database should be straightforward.

## Some additional requirements

1. We should re-publish the sitemap on an hourly cadence
2. We should ingest the json on a daily cadence.
3. CPC codes should always be updated to reflect the latest json
4. there is no need to store historic values for CPC codes.
5. CPC codes should be truncated to the first four characters to reduce false specificity

**Comment by slifty** [comment]
@metasj @isTravis do we have any word back about registering a developer URL for the integration with the google black box for this part of the process?

**Comment by metasj** [comment]
You mean a second URL for a dev sitemap?  Is there a way to avoid most of that parsing being duplicated work? 
I'll ask Ian.

**Comment by slifty** [comment]
@metasj yes -- it is my understanding that right now there is a v1 production sitemap that gets ingested and an endpoint that spits out the CPC json for that production sitemap; we need one so that we can test that pipeline on v2

**Comment by metasj** [comment]
Name the dev map and update this issue ~
Ian: "We could aim our staging pipeline at the dev sitemap and output to a separate file."

**Comment by slifty** [comment]
Implementing the sitemap generation code as a worker that can be triggered through a package.json script; I believe we are using Heroku, which means that we'll want to set up the [Heroku scheduler](https://devcenter.heroku.com/articles/scheduler) to call it on a regular (daily?) basis.

**Comment by metasj** [comment]
Ok, just let me know what to tell Ian and they'll set it up
